scripts/db_restore.sh

```diff
@@ -72,13 +74,50 @@ fi
   set -e
 } | tee >/dev/null
 
-# The first time we run it, we will trigger FK constraints errors
-set +e
-pg_restore -U postgres -h localhost --no-acl --no-owner --role=${LOCALDBUSER} -n public -O -c -d "${LOCALDBNAME}" "${DBDUMP_FILE}" 2>/dev/null
-set -e
+PG_RESTORE_OPTIONS=(-U postgres -h localhost --no-acl --no-owner --role="${LOCALDBUSER}" -n public -O -d "${LOCALDBNAME}")
+PG_RESTORE_COMMAND=pg_restore
+if [ -n "$USE_DOCKER" ]; then
+  PG_RESTORE_COMMAND="sudo docker run --rm --network host -v $(dirname "$DBDUMP_FILE"):/backup postgres:14 pg_restore -h localhost -p 5432 -U opencollective"
+  DBDUMP_FILE="/backup/$(basename "$DBDUMP_FILE")"
+fi
+
+# The first time we run it with the -s option (schema only)
+echo "Restoring schema"
+$PG_RESTORE_COMMAND "${PG_RESTORE_OPTIONS[@]}" -s "${DBDUMP_FILE}"
+
+# Disable triggers
+echo "Disabling triggers"
+psql -U postgres -h localhost "${LOCALDBNAME}" -c "
+DO \$\$ 
+BEGIN 
+   EXECUTE (
+      SELECT string_agg(
+         format('ALTER TABLE %I.%I DISABLE TRIGGER ALL', schemaname, tablename), '; '
+      ) 
+      FROM pg_tables 
+      WHERE schemaname = 'public'
+   ); 
+END \$\$;
+"
+
+# Restore data (-a flag)
+echo "Restoring data"
+$PG_RESTORE_COMMAND "${PG_RESTORE_OPTIONS[@]}" -a "${DBDUMP_FILE}"
 
-# So we run it twice :-)
-pg_restore -U postgres -h localhost --no-acl --no-owner --role=${LOCALDBUSER} -n public -O -c -d "${LOCALDBNAME}" "${DBDUMP_FILE}"
+# Re-enable triggers
+echo "Re-enabling triggers"
+psql -U postgres -h localhost "${LOCALDBNAME}" -c "
+DO \$\$ 
+BEGIN 
+   EXECUTE (
+      SELECT string_agg(
+         format('ALTER TABLE %I.%I ENABLE TRIGGER ALL', schemaname, tablename), '; '
+      ) 
+      FROM pg_tables 
+      WHERE schemaname = 'public'
+   ); 
+END \$\$;
+"
```

Comment:
Two changes here:
1. Rather than bruteforcing the same pg_restore command 2 times, we're now restoring only the schema first (`-s`) and then the data (`-a`).
2. Disabling triggers before restoring the data, to prevent any missing foreign key issues from popping in for interdependent rows.